pkgs/stdenv/generic/default.nix

```diff
@@ -402,13 +402,10 @@ let
 
       # Utility flags to test the type of platform.
       inherit (hostPlatform)
-        isDarwin isLinux isSunOS isCygwin isFreeBSD isOpenBSD isi686 isx86_64
-        is64bit isMips isBigEndian;
+        isDarwin isLinux isSunOS isHurd isCygwin isFreeBSD isOpenBSD
+        isi686 isx86_64 is64bit isMips isBigEndian;
       isArm = hostPlatform.isArm32;
       isAarch64 = hostPlatform.isArm64;
-      # Other code instead checks for anything using GNU userland,
-      # e.g. GNU/linux. This refers just to GNU Hurd.
-      isGNU = system == "i686-gnu";
```

Comment:
There might be a few references to this you missed:

- https://github.com/NixOS/nixpkgs/blob/9d5c28d22cf6573da2f75da2b9ec67f351c92f7d/pkgs/development/tools/misc/gdb/default.nix#L20-L22
- https://github.com/NixOS/nixpkgs/blob/9d5c28d22cf6573da2f75da2b9ec67f351c92f7d/pkgs/tools/archivers/zpaq/zpaqd.nix#L12